libraries/hotstuff/include/eosio/hotstuff/chain_pacemaker.hpp

```diff
@@ -54,7 +54,11 @@ namespace eosio::hotstuff {
       // These requests can come directly from the net threads, or indirectly from a
       //   dedicated finalizer thread (TODO: discuss).
 #warning discuss
-      std::mutex              _hotstuff_global_mutex;
+      mutable std::mutex      _hotstuff_global_mutex;
+
+      mutable finalizer_state     _state_cache;
+      mutable std::shared_mutex   _state_cache_mutex;
```

Comment:
Please move mutex above data it is protecting and add comment to what it is protecting.